src/GafferUI/AnnotationsGadget.cpp

```diff
@@ -319,6 +346,17 @@ void AnnotationsGadget::update() const
 		MetadataAlgo::annotations( node, names );
 		for( const auto &name : names )
 		{
+			if( !StringAlgo::matchMultiple( name, m_visibleAnnotations ) )
+			{
+				if(
+					name == "user" || binary_search( templates.begin(), templates.end(), name ) ||
+					!StringAlgo::matchMultiple( untemplatedAnnotations, m_visibleAnnotations )
```

Comment:
This logic is totally correct, but something about it caused me to stare at it for way too long to determine that.  Maybe it's just me, but I might be tempted to lift `bool untemplatedVisible = StringAlgo::matchMultiple( untemplatedAnnotations, m_visibleAnnotations )` out of the loop, and declare a variable `bool nameIsUntemplated = name != "user" && !binary_search( templates.begin(), templates.end(), name )`, so that this if just becomes `if( !( nameIsUntemplated && untemplatedVisible ) )`.  I dunno, maybe that looks worse to you, but something about having this expressed as an `or` feels weird, to me, the more direct understanding is "be visible when all these negated conditions all match".  Not important, just leaving comments to show I've read the code.

Reply:
Writing it did make my head hurt, but so does `if( !( nameIsUntemplated && untemplatedVisible ) )`. I've tried a slight variation on your suggestion in https://github.com/GafferHQ/gaffer/pull/4232/commits/1eb456c47e8ac8ee22d9f8d0e7ddc1d42abcc336.